src/main/java/captain/cybot/adventure/backend/model/user/User.java

```diff
@@ -19,18 +23,30 @@ public class User {
     @Id
     @GeneratedValue(strategy = GenerationType.AUTO)
     private Long id;
+
+    @NotBlank(message = "Username is required")
+    @Size(min = 6, max = 30, message = "Username must be between 6-30 characters in length")
     private String username;
+
+    @NotBlank(message = "Email is required")
+    @Email(message = "Email must be valid")
     private String email;
+
+
+    @NotBlank(message = "Password is required")
     private String password;
 
-    @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+    @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
     @JoinColumn(name = "cosmetic_id", referencedColumnName = "id")
     private Cosmetic cosmetic;
 
     @OneToMany(fetch = FetchType.LAZY, mappedBy = "user")
     @JsonManagedReference
     private List<World> worlds;
 
+    @ManyToMany(fetch =  FetchType.EAGER)
+    private Collection<Role> roles = new ArrayList<>();
```

Comment:
The role is added to the User using a setter. Not upon construction because the "/api/v0/users" API only takes in the following fields, not "roles":
{
    "username":"testuser",
    "email":"testuser@gmail.com",
    "password":"Secretpass123!"
}